### PR TITLE
adding xrt recipe

### PR DESCRIPTION
**Pull request by kbeeperez**
implements: https://github.com/nsls-ii-forge/xrt-feedstock

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kbeeperez** [comment]
Attn @mrakitin

**Review by mrakitin — APPROVED** [review]
Thanks, @kbeeperez and @dopplershift!

I agree to be a maintainer of this feedstock/recipe.

File: recipes/xrt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  skip: true  # [py<36]
+  entry_points:
+    - {{ name }} = {{ name }}.gui.xrtQookStart:main
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (dopplershift) [inline recipes/xrt/meta.yaml:24]: I don't see anywhere this is using a compiler to install the package...in fact, I think this could be `noarch: python`?

Comment (mrakitin) [reply recipes/xrt/meta.yaml:24]: You are right it's not building C/C++ extensions. The package uses .cl files, that can probably be handled by the addition of [`pyopencl`](https://github.com/conda-forge/pyopencl-feedstock/blob/master/recipe/meta.yaml) dependency.

Comment (mrakitin) [inline recipes/xrt/meta.yaml:24]:
```suggestion
```

File: recipes/xrt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  host:
+    - python
+    - pip
+  run:
+    - python
```
Comment (mrakitin) [inline recipes/xrt/meta.yaml:29]:
```suggestion
    - python >=3.6
```

File: recipes/xrt/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  skip: true  # [py<36]
+  entry_points:
+    - {{ name }} = {{ name }}.gui.xrtQookStart:main
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
```
Comment (mrakitin) [inline recipes/xrt/meta.yaml:26]:
```suggestion
    - python >=3.6
```

File: recipes/xrt/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.zip
+  sha256: 19c0d85357019d91b324c5ba5ffe7cc6fb31e70f1fa2c9c869d1ee6b63a7338f
+  # TODO: remove the patch after https://github.com/kklmn/xrt/pull/42 is accepted/merged/released.
+  patches:
+    - entry-points.patch
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  skip: true  # [py<36]
```
Comment (mrakitin) [inline recipes/xrt/meta.yaml:18]:
```suggestion
  noarch: python
```

File: recipes/xrt/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - matplotlib
+    - numpy
+    - pyqt
```
Comment (mrakitin) [inline recipes/xrt/meta.yaml:32]:
```suggestion
    - pyopencl
    - pyqt
```

File: recipes/xrt/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    - scipy
+    - sphinx
+    # - pyopencl  - Not sure if we need this (optional runtime dep)
```
Comment (mrakitin) [inline recipes/xrt/meta.yaml:35]:
```suggestion
```

**Review by mrakitin — COMMENTED** [review]
Here are my suggestions for @kbeeperez to convert it to the `noarch` package. I added the `pyopencl` run dependency.

File: recipes/xrt/meta.yaml (hunk 1 above)
Comment (mrakitin) [reply recipes/xrt/meta.yaml]: This is done, the builds are passing now.

**Review by dopplershift — APPROVED** [review]

**Comment by mrakitin** [comment]
@dopplershift, thanks for the useful suggestion about converting to `noarch`, we'll see how it survives the tests on our side.

**Comment by mrakitin** [comment]
@dopplershift, @conda-forge/staged-recipes, I noticed the failed build after the feedstock was created - it failed to apply the patch. I checked the diffs between the original patch added here, and the one that ended up in the feedstock. They appeared to be different - in particular, the one in the feedstock did not have the CRLF line endings, as the original source was written in Windows, but the feedstock's version appeared to be "sanitized" (I believe). Is it possible to preserve the original patch "as is" in the future? Or is it something to do with the OS where the bot is operating in?

## Steps to reproduce:
```bash
$ mkdir -p ~/tmp/xrt-patches-diff && cd ~/tmp/xrt-patches-diff
$ wget https://raw.githubusercontent.com/kbeeperez/staged-recipes/cb160eb3300fd0796284fb44099bd145be59b816/recipes/xrt/entry-points.patch -O entry-points-staged-recipes.patch
$ wget https://raw.githubusercontent.com/conda-forge/xrt-feedstock/master/recipe/entry-points.patch -O entry-points-xrt-feedstock.patch
$ git diff entry-points-staged-recipes.patch entry-points-xrt-feedstock.patch
```

```diff
diff --git a/entry-points-staged-recipes.patch b/entry-points-xrt-feedstock.patch
index 72974af..4cef307 100644
--- a/entry-points-staged-recipes.patch
+++ b/entry-points-xrt-feedstock.patch
@@ -3,31 +3,31 @@ index 50d17f2..97a5a75 100644
 --- a/setup.py
 +++ b/setup.py
 @@ -204,6 +204,7 @@ setup(
-         'xrt.gui.xrtQook': ['_icons/*.*'],
-         'xrt.gui.xrtGlow': ['_icons/*.*']},
-     scripts=['xrt/gui/xrtQookStart.pyw', 'xrt/gui/xrtQookStart.py'],
-+    entry_points={'console_scripts': ['xrt = xrt.gui.xrtQookStart:main']},
-     install_requires=['numpy>=1.8.0', 'scipy>=0.17.0', 'matplotlib>=2.0.0',
-                       'sphinx>=1.6.2'],
-     classifiers=['Development Status :: 5 - Production/Stable',
+         'xrt.gui.xrtQook': ['_icons/*.*'],
+         'xrt.gui.xrtGlow': ['_icons/*.*']},
+     scripts=['xrt/gui/xrtQookStart.pyw', 'xrt/gui/xrtQookStart.py'],
++    entry_points={'console_scripts': ['xrt = xrt.gui.xrtQookStart:main']},
+     install_requires=['numpy>=1.8.0', 'scipy>=0.17.0', 'matplotlib>=2.0.0',
+                       'sphinx>=1.6.2'],
+     classifiers=['Development Status :: 5 - Production/Stable',
 diff --git a/xrt/gui/xrtQookStart.py b/xrt/gui/xrtQookStart.py
 index d998978..346af9b 100644
 --- a/xrt/gui/xrtQookStart.py
 +++ b/xrt/gui/xrtQookStart.py
 @@ -9,7 +9,7 @@ sys.path.append(os.path.join('..', '..'))
- import xrt.gui.xrtQook as xQ
-
-
--if __name__ == '__main__':
-+def main():
-     if any('spyder' in name.lower() for name in os.environ):
-         pass  # spyder is present
-     else:
+ import xrt.gui.xrtQook as xQ
+
+
+-if __name__ == '__main__':
++def main():
+     if any('spyder' in name.lower() for name in os.environ):
+         pass  # spyder is present
+     else:
 @@ -29,3 +29,7 @@ if __name__ == '__main__':
-     ex.setWindowTitle("xrtQook")
-     ex.show()
-     sys.exit(app.exec_())
-+
-+
-+if __name__ == '__main__':
-+    main()
+     ex.setWindowTitle("xrtQook")
+     ex.show()
+     sys.exit(app.exec_())
++
++
++if __name__ == '__main__':
++    main()
```

The following command with `--ignore-cr-at-eol` returns a zero diff:
```bash
$ git diff --ignore-cr-at-eol entry-points-staged-recipes.patch entry-points-xrt-feedstock.patch
```

```bash
$ git diff --help
...
       --ignore-cr-at-eol
           Ignore carrige-return at the end of line when doing a comparison.
...
```

After I corrected the patch in the feedstock via https://github.com/conda-forge/xrt-feedstock/pull/1/commits/fecabb4fefe071f6d2c720fc8e25578db61984aa, the build passed. Looking forward for your comments!

**Comment by dopplershift** [comment]
No clue. @ocefpaf any thoughts?

**Pull request merged**